### DISCUSSION TITLE
Sound Toggle?

### DISCUSSION THREAD

**Discussion by Durahl** [EdgeTX/edgetx#3661] [Q&A]
Greetings!

I'd like to implement detailed Audio System where one Switch Position activates the same Setting **_but_** toggles between two Audio Files to reflect the State the System is in.

**To explain it in more Detail:**
- The Controllers for the Models I use are as dumb as they can get without any Telemetry Features.
- Performing an Action requires for a Channel to be set to either end of the Channel Range - Like 1000 or 2000.
- An Action enabled by setting it to 2000 will not be disabled by setting it to 1000, only setting it to 2000 _again_ will do so.

Programming a Switch to work like that is not the Problem - I already got that covered.

The Problem is for the Audio announcing the State to toggle between the two States of something changing:
- Using the Momentary SH Switch as an Example if I activate it, I want the Lights on the Vehicle to come `ON` and the Remote to say so with a "`LightsON.wav`"
- Using the same SH Switch if I activate it again, I want the Lights on the Vehicle to turn `OFF` and the Remote to say so with a "`LightsOFF.wav`"

Since I don't know how to have the Sounds toggle between each state, I'm currently only having the Remote say "`Lights.wav`" 🥲

I do realize that with the Controller not giving any feedback of the State something is there is a possibility of the two coming out of sync but in all honesty I cannot see that ever really happening and quite frankly it ever happening would also not be the end of the world.

Can this be done and how? ( please explain including Screenshots )

**Comment by jrwieland** [comment]
it can be accomplished
Using LS to trigger a SF
Screen shots attached
Use edge on the LS and set 2 values 
(1) for a moment switch depress and let go
(2) a slightlly longer hold of the switch results in the second value triggering
Play with the values for how long you want to hold the Momentary switch to activate LS2 in the example
![LS Switches](https://github.com/EdgeTX/edgetx/assets/20526716/c4b4d6e0-9961-4a84-93b4-9d9221bfbb10)
![SF Switches](https://github.com/EdgeTX/edgetx/assets/20526716/9a978120-8889-4b46-90df-12574b7afe2e)

**Comment by pfeerick** [comment]
I mentioned this on Discord, but am posting it again here for completeness:

It's possible to do this with more complicated logic, such as more logical switches, and a global variable which could make tracking the A/B state a little more reliable, but I think this should suffice. Using a sticky logical switch will "remember" the state of the toggle switch - where V1 sets it, and V2 resets it (similar to your lights). When you press the switch the first time, it will turn the LS on, and when you press it the second time, it will turn the LS off. You can then have two special functions, one that triggers when L01 is active (L01), and one when LO1 is not active (!L01). By setting the repeat to 1x, it only plays the sound once (rather than continuously/repeatedly). Better still, set it to !1x (it's the very first option), and it also won't play when you turn the radio on or switch models if the switch is held down. 

I have a feeling this may also be relevant for some indoor flyer models that have lights - I'm sure I've seen mention of one that behaves the same as this... i.e. only changes states on the highs.

Logical switch
![image](https://github.com/EdgeTX/edgetx/assets/5500713/94d17f88-fc59-44eb-9acd-bcb603578a16)

Special functions
![image](https://github.com/EdgeTX/edgetx/assets/5500713/39241a1d-29c1-4130-a112-01b869ea01fd)
![image](https://github.com/EdgeTX/edgetx/assets/5500713/ac6944b1-9fc6-40fa-baea-879352413f73)
